src/rgw/driver/rados/rgw_d3n_datacache.h

```diff
@@ -51,14 +51,27 @@ struct D3nCacheAioWriteRequest {
 	int d3n_prepare_libaio_write_op(bufferlist& bl, unsigned int len, std::string oid, std::string cache_location);
 
   ~D3nCacheAioWriteRequest() {
-    ::close(fd);
+		::close(fd);
 		cb->aio_buf = nullptr;
 		free(data);
 		data = nullptr;
 		delete(cb);
   }
 };
 
+struct D3nFileClose {
```

Comment:
why do we need this in the header file?
it is only used inside rthe cpp file

Reply:
you may choose to use a lambda instead of a function object.
you can pass `r` by reference to the lambda and allow for the behavior where the return value from the function reflects a possible failure to close the file